### PR TITLE
goimports

### LINKED ISSUE #16621: Add goimports lint check to .golangci.yml
#### Summary
This ticket is about adding the goimports lint check to the repository. The way to go about it would be to add

```diff
diff --git a/.golangci.yml b/.golangci.yml
index 71c041aa5..a6a841d43 100644
--- a/.golangci.yml
+++ b/.golangci.yml
@@ -10,6 +10,8 @@ linters-settings:
   govet:
     check-shadowing: true
     enable-all: true
+  goimports:
+    local-prefixes: github.com/mattermost/mattermost-server/v5
 
 linters:
   disable-all: true
@@ -27,6 +29,7 @@ linters:
     - unused
     - varcheck
     - misspell
+    - goimports
     # TODO: enable this later
     # - errcheck
```

in the .golangci.yml file. 

And then fix all issues that arise until `make golangci-lint` passes with no errors.

### PR DESCRIPTION
**Pull request by agnivade**
- format using `goimports -local github.com/mattermost/mattermost-server/v5 -w`
- added goimports lint check to .golangci.yml
- format using `goimports -local github.com/mattermost/mattermost-server/v5 -w` for a corner case
- make app-layers, *-mocks and store-layers for ci check

Fixes #16621

### CODE REVIEW

**Review by nronas — APPROVED** [review]
LGTM

File: app/import_types.go [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
 package app
 
-import "github.com/mattermost/mattermost-server/v5/model"
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+)
```
Comment (isacikgoz) [inline app/import_types.go:8]: Hmm, wasn't expecting this 😅

**Review by isacikgoz — APPROVED** [review]
LGTM, surprised that we use parenthesizes for a single import.

**Pull request merged**